pkg/policy/api/rule.go

```diff
@@ -61,3 +61,30 @@ type Rule struct {
 	// +optional
 	Description string `json:"description,omitempty"`
 }
+
+// HasChildRule it return true if the rule has a child rule.
+func (r *Rule) HasChildRule() bool {
+	for _, rule := range r.Egress {
+		if rule.HasChildRule() {
+			return true
+		}
+	}
+	return false
+}
+
+// CreateChildRule will return a new Rule with the new data based gather
+// by the rules that autogenerated new Rule
+func (r *Rule) CreateChildRule() (*Rule, error) {
+	newRule := r.DeepCopy()
+	newRule.Egress = []EgressRule{}
+	newRule.Ingress = []IngressRule{}
```

Comment:
This is definitely incorrect, it will zero out the ingress rule section without recreating it.